src/game.h

```diff
@@ -419,16 +419,20 @@ class Game
 	void updateCreatureWalkthrough(const Creature* creature);
 	void updateKnownCreature(const Creature* creature);
 
-	GameState_t getGameState() const;
-	void setGameState(GameState_t newState);
-	void saveGameState();
+	template <GameState_t S>
```

Comment:
there is no point in using a template in this case, we should avoid that if not necessary

Reply:
just use a classic argument of type GameState_t and that's it, why add so much magic for a simple return int == int;